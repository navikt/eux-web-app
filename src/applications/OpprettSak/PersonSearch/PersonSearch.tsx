import { Alert, Loader, Search } from '@navikt/ds-react'
import { Person } from 'declarations/types'
import _ from 'lodash'
import { PileDiv, VerticalSeparatorDiv } from '@navikt/hoykontrast'
import PT from 'prop-types'
import React, { useCallback, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'

export interface PersonSearchProps {
  alertMessage: JSX.Element | string | undefined
  alertType: string | undefined
  alertTypesWatched: Array<string> | undefined
  className?: string
  error: string | undefined
  id ?: string
  initialFnr: any
  parentNamespace: string
  searchingPerson: boolean
  onFnrChange?: (newFnr: string) => void
  onPersonFound?: (person: Person) => void
  onSearchPerformed: (fnr: any) => void
  person?: Person | null | undefined
  value: string | undefined
}

const PersonSearch: React.FC<PersonSearchProps> = ({
  alertMessage,
  alertType,
  alertTypesWatched = [],
  error,
  id,
  initialFnr,
  parentNamespace,
  searchingPerson,
  onFnrChange,
  onPersonFound,
  onSearchPerformed,
  person,
  value
}: PersonSearchProps): JSX.Element => {
  const { t } = useTranslation()
  const [fnr, setFnr] = useState<string>(initialFnr ?? value)
  const [_person, setPerson] = useState<Person | null | undefined>(undefined)
  const [localValidation, setLocalValidation] = useState<string | undefined>(undefined)
  const namespace = parentNamespace + '-personSearch'

  useEffect(() => {
    setFnr(value ?? initialFnr)
  }, [value])

  const isPersonValid = useCallback((person: Person) =>
    person?.fornavn?.length !== undefined && person?.fnr !== undefined,
  []
  )

  useEffect(() => {
    if (person && !_person && isPersonValid(person)) {
      setPerson(person)
      if (_.isFunction(onPersonFound)) {
        onPersonFound(person)
      }
    }
  }, [person])

  const onSearch = (fnr: string | number | readonly string[]) => {
    if (!fnr) {
      setLocalValidation(t('validation:noFnr'))
      return
    }
    const fnrPattern = /^[0-9]{11}$/
    if (fnr && !fnrPattern.test('' + fnr)) {
      setLocalValidation(t('validation:invalidFnr'))
      return
    }
    setLocalValidation(undefined)
    setPerson(undefined)
    if (_.isFunction(onSearchPerformed)) {
      onSearchPerformed(fnr)
    }
  }

  const onChange = (fnr: string): void => {
    setLocalValidation(undefined)
    const newFnr = fnr.trim()
    setFnr(newFnr)
    if (_.isFunction(onFnrChange)) {
      onFnrChange(newFnr)
    }
  }

  return (
    <PileDiv style={{ alignItems: 'flex-start' }}>
      <Search
        label={t('label:sÃ¸ker')}
        /* error={error ?? localValidation} */
        data-testid={id ?? namespace + '-saksnummerOrFnr'}
        id={id ?? namespace + '-saksnummerOrFnr'}
        onChange={onChange}
        required
        hideLabel={false}
        value={fnr || ''}
        disabled={searchingPerson}
        onSearch={onSearch}
      >
        <Search.Button>
          {searchingPerson ? t('message:loading-searching') : t('el:button-search')}
          {searchingPerson && <Loader />}
        </Search.Button>
      </Search>
      {(error ?? localValidation) && (
        <>
          <VerticalSeparatorDiv size='0.5' />
          <span className='navds-error-message navds-error-message--medium'>
            {error ?? localValidation}
          </span>
        </>
      )}
      {alertMessage && alertType && alertTypesWatched.indexOf(alertType) >= 0 && (
        <Alert variant='warning'>
          {alertMessage}
        </Alert>
      )}
    </PileDiv>
  )
}

PersonSearch.propTypes = {
  className: PT.string,
  onFnrChange: PT.func,
  onPersonFound: PT.func
  // validation: ErrorElementPropType
}

export default PersonSearch
