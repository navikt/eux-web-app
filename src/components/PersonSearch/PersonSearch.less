@import 'src/constants';

.personsok {
  margin-bottom: 2em;

  .personsok__skjema {
    display: flex;
    flex-direction: row;
  }
  .personsok__input {
    min-width: 24.5em;
  }
  .personsok__spinnerwrapper {
    margin: 2.3em 1em 0 -2.5em;
  }

  .personsok__knapp {
    display: flex;
    flex: 0;
    height: 2.4em;
    align-self: flex-start;
    margin: 1.9em 0 0 1em;
  }

  .personsok__kort {
    margin: 1em 0;
  }

  .personsok__advarsel {
    margin-top: 1em;
  }
  .familierelasjoner__knapp {
    display: flex;
    align-self: center;
    justify-self: flex-end;

    .familierelasjoner__knapp__label {
      display: flex;
      align-self: center;
    }
  }
  .familierelasjoner__knapp__ikon {
    margin: -0.0em 0.5em 0 0;

    path {
      stroke: @navBla;
    }
  }
  .knapp--disabled {
    path {
      stroke: @white !important;
    }
  }

  .familierelasjoner__knapp {

    &:hover:not(.knapp--disabled) {
      .familierelasjoner__knapp__ikon path {
        stroke: @white !important;
      }
    }
  }
}
