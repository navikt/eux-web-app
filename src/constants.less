@import './variabler';

@black: #000000;
@font-color: @black;
