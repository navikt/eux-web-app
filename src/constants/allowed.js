import {IS_DEVELOPMENT} from "constants/environment.ts";

export const ALLOWED_TO_FILL_OUT = IS_DEVELOPMENT ? ["H001", "F001"] : ["H001"]
export const ALLOWED_SAKSHANDLINGER = IS_DEVELOPMENT ?
    ["H001", "X001", "X009", "F002", "F004", "F022", "F023", "F026", "F027", "F016", "R001", "R003", "R004", "Delete_Case"] :
    ["H001", "X001", "X009", "F002", "Delete_Case"]

export const HIDDEN_SAKSHANDLINGER = ["multipleParticipants", "singleParticipant"]
export const ALLOWED_SED_HANDLINGER = ["H002", "F002", "X008", "X010", "X011", "X012", ]
export const ALLOWED_SED_EDIT_AND_UPDATE = ["H001", "H002", "F002", "X008", "X010", "X011", "X012"]
