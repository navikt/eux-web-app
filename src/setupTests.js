/* eslint-disable */

import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import each from 'jest-each';

global.fetch = require('jest-fetch-mock');

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.mount = mount;

global.each = each;
