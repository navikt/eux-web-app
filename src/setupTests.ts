import {render, screen} from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import { useDispatch, useSelector } from 'react-redux'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'

jest.mock('react-redux');

(global as any).screen = screen;
(global as any).render = render;
(global as any).act = act

// eslint-disable-next-line no-undef
HTMLCanvasElement.prototype.getContext = jest.fn()
window.scrollTo = jest.fn()

Object.defineProperty(window, 'getComputedStyle', {
  value: () => ({
    getPropertyValue: () => ('')
  })
})

jest.mock('amplitude-js', () => ({
  getInstance: () => ({
    init: jest.fn(),
    logEvent: jest.fn()
  })
}))

jest.mock('i18next', () => {
  const use = jest.fn()
  const init = jest.fn()
  const loadLanguages = jest.fn()
  const result = {
    use: use,
    init: init,
    loadLanguages: loadLanguages
  }
  use.mockImplementation(() => result)
  return result
})

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

export const stageSelector = (defaultSelector: any, params: any) => {
  (useDispatch as jest.Mock).mockImplementation(() => jest.fn());
  (useSelector as jest.Mock).mockImplementation(() => ({
    ...defaultSelector,
    ...params
  }))
}
