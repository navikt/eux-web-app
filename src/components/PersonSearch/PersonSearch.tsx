import Alert, { AlertStatus } from 'components/Alert/Alert'
import WaitingPanel from 'components/WaitingPanel/WaitingPanel'
import * as types from 'constants/actionTypes'
import { Person } from 'declarations/types'
import { ValidationPropType } from 'declarations/types.pt'
import _ from 'lodash'
import { Knapp } from 'nav-frontend-knapper'
import { Input } from 'nav-frontend-skjema'
import PT from 'prop-types'
import React, { useCallback, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components'
import PersonCard from '../PersonCard/PersonCard'

export interface PersonSearchProps {
  alertStatus: string | undefined;
  alertMessage: string | undefined;
  alertType: string | undefined;
  className?: string;
  initialFnr: any;
  gettingPerson: boolean;
  onAlertClose: () => void;
  onFnrChange?: () => void;
  onPersonFound? : (person: Person) => void;
  onSearchPerformed: (fnr: any) => void;
  onPersonRemoved: () => void;
  person?: Person;
  resetAllValidation: () => void;
  validation: any;
}

const AlertstripeDiv = styled.div`
  margin: 0.5rem;
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
  width: 50%;
`
const PersonSearchDiv = styled.div`
  margin-bottom: 2em;
`
const PersonSearchPanel = styled.div`
  display: flex;
  flex-direction: row;
`
const PersonSearchInput = styled(Input)`
  min-width: 24.5em;
`
const Button = styled(Knapp)`
  display: flex;
  flex: 0;
  height: 2.4em;
  align-self: flex-start;
  margin: 1.9em 0 0 1em;
 `

const PersonSearch: React.FC<PersonSearchProps> = ({
  alertStatus, alertMessage, alertType, className, initialFnr, gettingPerson,
  onAlertClose, onFnrChange, onPersonFound, onPersonRemoved, onSearchPerformed, person,
  resetAllValidation, validation
}: PersonSearchProps): JSX.Element => {
  const { t } = useTranslation()
  const [_fnr, setFnr] = useState<string | undefined>(initialFnr)
  const [_person, setPerson] = useState<Person | undefined>(undefined)
  const [localValidation, setLocalValidation] = useState<string |undefined>(undefined)

  const isPersonValid = useCallback(
    (person: Person) => (person?.fornavn?.length !== undefined && person?.fnr !== undefined)
    , []
  )

  useEffect(() => {
    if (person && !_person && isPersonValid(person)) {
      setPerson(person)
      if (_.isFunction(onPersonFound)) {
        onPersonFound(person)
      }
    }
  }, [person, _person, isPersonValid, onPersonFound])

  const sokEtterPerson = (): void => {
    const fnrPattern = /^[0-9]{11}$/
    if (_fnr && !fnrPattern.test(_fnr)) {
      setLocalValidation(t('ui:validation-invalidFnr'))
      return
    }
    setLocalValidation(undefined)
    setPerson(undefined)
    if (_.isFunction(onSearchPerformed)) {
      onSearchPerformed(_fnr)
    }
  }

  const onChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setLocalValidation(undefined)
    resetAllValidation()
    if (_.isFunction(onFnrChange)) {
      onFnrChange()
    }
    setFnr(e.target.value.trim())
  }

  const onRemovePerson = (): void => {
    setLocalValidation(undefined)
    resetAllValidation()
    setPerson(undefined)
    if (_.isFunction(onPersonRemoved)) {
      onPersonRemoved()
    }
  }

  return (
    <PersonSearchDiv>
      <PersonSearchPanel>
        <PersonSearchInput
          data-testid='personsok__input-id'
          label={t('ui:form-searchUser')}
          value={_fnr || ''}
          onChange={onChange}
          feil={validation.fnr || localValidation}
        />
        <Button onClick={sokEtterPerson} disabled={gettingPerson}>
          {gettingPerson ? <WaitingPanel size='S' message={t('ui:form-searching')} oneLine /> : t('ui:form-search')}
        </Button>
      </PersonSearchPanel>
      {alertMessage && alertType === types.SAK_PERSON_GET_FAILURE && (
        <AlertstripeDiv>
          <Alert
            type='client'
            fixed={false}
            message={t(alertMessage)}
            status={alertStatus as AlertStatus}
            onClose={onAlertClose}
          />
        </AlertstripeDiv>
      )}
      {person && isPersonValid(person) && <PersonCard className='neutral' person={person} onRemoveClick={onRemovePerson} />}
    </PersonSearchDiv>
  )
}

PersonSearch.propTypes = {
  className: PT.string,
  onFnrChange: PT.func,
  onPersonFound: PT.func,
  resetAllValidation: PT.func.isRequired,
  validation: ValidationPropType.isRequired
}

export default PersonSearch
