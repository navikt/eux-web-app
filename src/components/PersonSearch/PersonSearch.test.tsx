import * as types from 'constants/actionTypes'
import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import samplePerson from 'resources/tests/samplePerson'
import { stageSelector } from 'setupTests'
import PersonSearch, { PersonSearchProps, PersonSearchSelector } from './PersonSearch'

describe('components/PersonSearch/PersonSearch', () => {
  let wrapper: ReactWrapper
  const initialMockProps: PersonSearchProps = {
    className: 'mock-className',
    onFnrChange: jest.fn(),
    onPersonFound: jest.fn(),
    resetAllValidation: jest.fn(),
    validation: {}
  }

  const defaultSelector: PersonSearchSelector = {
    alertStatus: 'ERROR',
    alertMessage: 'message',
    alertType: types.SAK_PERSON_GET_FAILURE,
    fnr: '12345678901',
    gettingPerson: false,
    person: samplePerson
  }

  beforeAll(() => {
    stageSelector(defaultSelector, {})
  })

  beforeEach(() => {
    wrapper = mount(<PersonSearch {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('HTML structure', () => {
    expect(wrapper.exists('.personsok')).toBeTruthy()
  })
})
