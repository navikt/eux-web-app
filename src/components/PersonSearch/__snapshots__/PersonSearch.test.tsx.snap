// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PersonSearch/PersonSearch Render: match snapshot 1`] = `
.c5 {
  opacity: 0;
  -webkit-animation: cJoqxJ 0.5s forwards;
  animation: cJoqxJ 0.5s forwards;
  position: relative;
}

.c5 .alertstripe__tekst {
  -webkit-flex: auto !important;
  -ms-flex: auto !important;
  flex: auto !important;
  margin: auto 0;
  max-width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 .fixed {
  position: fixed;
  top: 0.25rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  z-index: 10;
}

.c5 .closeIcon {
  position: absolute;
  top: 0.25rem;
  right: 0.25rem;
  cursor: pointer;
}

.c5 .type-server {
  border-radius: 0px !important;
  border: 0px !important;
}

.c5 .type-server .alertstripe__tekst {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: none !important;
}

.c6 {
  position: absolute;
  top: 0.25rem;
  right: 0.25rem;
  cursor: pointer;
}

.c10 {
  margin-left: 1rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  border: 1px solid lightgray;
  background: transparent;
  margin-top: 1.5rem;
  max-width: 800px;
  min-width: 400px;
  padding: 1rem;
  border-radius: 5px;
}

.c7.personNotSelected {
  border: 3px solid red;
}

.c7.personSelected {
  border: 3px solid green;
}

.c7.neutral {
  background: white;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: flex-end;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4 {
  margin: 0.5rem;
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
  width: 50%;
}

.c0 {
  margin-bottom: 2em;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  min-width: 24.5em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  height: 2.4em;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin: 1.9em 0 0 1em;
}

<PersonSearch
  alertMessage="message"
  alertStatus="ERROR"
  alertType="SAK/PERSON/GET/FAILURE"
  alertTypesWatched={
    Array [
      "SAK/PERSON/GET/FAILURE",
    ]
  }
  className="mock-className"
  gettingPerson={false}
  initialFnr="12345678901"
  onAlertClose={[MockFunction]}
  onFnrChange={[MockFunction]}
  onPersonFound={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "etternavn": "BLYANT",
            "fdato": "1961-02-02",
            "fnr": "02026100715",
            "fornavn": "STOR",
            "kjoenn": "M",
            "relasjoner": Array [
              Object {
                "etternavn": "BLYANT",
                "fdato": "1961-04-08",
                "fnr": "02026100714",
                "fornavn": "PEN",
                "kjoenn": "K",
                "rolle": "EKTE",
              },
              Object {
                "etternavn": "BLYANT",
                "fdato": "1980-02-02",
                "fnr": "02026100713",
                "fornavn": "LITEN",
                "kjoenn": "M",
                "rolle": "BARN",
              },
            ],
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onPersonRemoved={[MockFunction]}
  onSearchPerformed={[MockFunction]}
  person={
    Object {
      "etternavn": "BLYANT",
      "fdato": "1961-02-02",
      "fnr": "02026100715",
      "fornavn": "STOR",
      "kjoenn": "M",
      "relasjoner": Array [
        Object {
          "etternavn": "BLYANT",
          "fdato": "1961-04-08",
          "fnr": "02026100714",
          "fornavn": "PEN",
          "kjoenn": "K",
          "rolle": "EKTE",
        },
        Object {
          "etternavn": "BLYANT",
          "fdato": "1980-02-02",
          "fnr": "02026100713",
          "fornavn": "LITEN",
          "kjoenn": "M",
          "rolle": "BARN",
        },
      ],
    }
  }
  resetAllValidation={[MockFunction]}
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <Styled(Input)
            label="ui:form-searchUser"
            onChange={[Function]}
            value="12345678901"
          >
            <Input
              bredde="fullbredde"
              className="c2"
              label="ui:form-searchUser"
              onChange={[Function]}
              value="12345678901"
            >
              <div
                className="skjemaelement c2"
              >
                <Label
                  htmlFor="182161559-2855-8941-14103-9113304175516"
                >
                  <label
                    className="skjemaelement__label"
                    htmlFor="182161559-2855-8941-14103-9113304175516"
                  >
                    ui:form-searchUser
                  </label>
                </Label>
                <input
                  aria-invalid={false}
                  className="skjemaelement__input input--fullbredde"
                  id="182161559-2855-8941-14103-9113304175516"
                  onChange={[Function]}
                  type="text"
                  value="12345678901"
                />
              </div>
            </Input>
          </Styled(Input)>
          <Styled(Knapp)
            disabled={false}
            onClick={[Function]}
          >
            <Knapp
              className="c3"
              disabled={false}
              onClick={[Function]}
            >
              <KnappBase
                autoDisableVedSpinner={false}
                className="c3"
                disabled={false}
                htmlType="submit"
                inaktivKlasseVedDisabled={false}
                kompakt={false}
                mini={false}
                onClick={[Function]}
                spinner={false}
                type="standard"
              >
                <button
                  className="knapp c3 knapp--standard"
                  disabled={false}
                  onClick={[Function]}
                  type="submit"
                >
                  ui:form-search
                </button>
              </KnappBase>
            </Knapp>
          </Styled(Knapp)>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="c4"
        >
          <Alert
            fixed={false}
            message="message"
            onClose={[MockFunction]}
            status="ERROR"
            type="client"
          >
            <Styled(AlertStripe)
              className="type-client status-ERROR"
              role="alert"
              type="feil"
            >
              <AlertStripe
                className="c5 type-client status-ERROR"
                form="default"
                role="alert"
                size="1.5em"
                type="feil"
              >
                <div
                  className="alertstripe c5 type-client status-ERROR alertstripe--feil"
                  role="alert"
                >
                  <span
                    className="alertstripe__ikon"
                  >
                    <span
                      className="sr-only"
                    >
                      feil
                    </span>
                    <Icon
                      kind="feil-sirkel-fyll"
                      size="1.5em"
                    >
                      <svg
                        focusable="false"
                        height="1.5em"
                        kind="feil-sirkel-fyll"
                        viewBox="0 0 24 24"
                        width="1.5em"
                      >
                        <g
                          fill="none"
                          fillRule="evenodd"
                        >
                          <path
                            d="M11.999 0C5.395 0 .013 5.372 0 11.976a11.923 11.923 0 0 0 3.498 8.493A11.925 11.925 0 0 0 11.977 24H12c6.603 0 11.986-5.373 12-11.978C24.013 5.406 18.64.012 11.999 0z"
                            fill="#A13A28"
                            fillRule="nonzero"
                          />
                          <path
                            d="M12 10.651l3.372-3.372a.954.954 0 1 1 1.349 1.35L13.349 12l3.372 3.372a.954.954 0 1 1-1.35 1.349L12 13.349 8.628 16.72a.954.954 0 1 1-1.349-1.35L10.651 12 7.28 8.628A.954.954 0 1 1 8.63 7.28L12 10.651z"
                            fill="#FFF"
                            fillRule="nonzero"
                          />
                        </g>
                      </svg>
                    </Icon>
                  </span>
                  <Normaltekst
                    className="alertstripe__tekst"
                    tag="div"
                  >
                    <TypografiBase
                      className="alertstripe__tekst"
                      tag="div"
                      type="normaltekst"
                    >
                      <div
                        className="typo-normal alertstripe__tekst"
                      >
                        message
                        <Styled(_default)
                          data-test-id="c-alert__close-icon"
                          onClick={[Function]}
                        >
                          <_default
                            className="c6"
                            data-test-id="c-alert__close-icon"
                            onClick={[Function]}
                          >
                            <svg
                              className="c6"
                              contentScriptType="text/ecmascript"
                              contentStyleType="text/css"
                              data-test-id="c-alert__close-icon"
                              enableBackground="new 0 0 24 24"
                              height={24}
                              id="Filled_Version"
                              onClick={[Function]}
                              preserveAspectRatio="xMidYMid meet"
                              version="1.1"
                              viewBox="0 0 24 24"
                              width={24}
                              x="0px"
                              xmlns="http://www.w3.org/2000/svg"
                              y="0px"
                              zoomAndPan="magnify"
                            >
                              <path
                                d="M11.499,0C5.17,0,0.012,5.148,0,11.477c-0.006,3.072,1.184,5.962,3.352,8.139c2.168,2.176,5.054,3.378,8.126,3.384H11.5  c6.328,0,11.487-5.149,11.5-11.479C23.012,5.181,17.863,0.012,11.499,0z M11.5,22.5L11.5,22.5L11.5,22.5L11.5,22.5z M16.088,15.398  c0.194,0.195,0.193,0.512-0.002,0.707c-0.098,0.098-0.225,0.146-0.353,0.146c-0.127,0-0.256-0.049-0.354-0.146l-3.882-3.898  l-3.896,3.882c-0.098,0.098-0.226,0.146-0.354,0.146c-0.128,0-0.255-0.049-0.353-0.146c-0.195-0.196-0.195-0.513,0.001-0.707  l3.897-3.882L6.91,7.602C6.715,7.406,6.716,7.09,6.912,6.896C7.107,6.699,7.424,6.7,7.619,6.897l3.881,3.897l3.896-3.881  c0.195-0.195,0.513-0.195,0.707,0.001c0.195,0.195,0.195,0.512-0.001,0.707l-3.896,3.882L16.088,15.398z"
                                fill="#3e3832"
                              />
                            </svg>
                          </_default>
                        </Styled(_default)>
                      </div>
                    </TypografiBase>
                  </Normaltekst>
                </div>
              </AlertStripe>
            </Styled(AlertStripe)>
          </Alert>
        </div>
      </styled.div>
      <PersonCard
        className="neutral"
        onRemoveClick={[Function]}
        person={
          Object {
            "etternavn": "BLYANT",
            "fdato": "1961-02-02",
            "fnr": "02026100715",
            "fornavn": "STOR",
            "kjoenn": "M",
            "relasjoner": Array [
              Object {
                "etternavn": "BLYANT",
                "fdato": "1961-04-08",
                "fnr": "02026100714",
                "fornavn": "PEN",
                "kjoenn": "K",
                "rolle": "EKTE",
              },
              Object {
                "etternavn": "BLYANT",
                "fdato": "1980-02-02",
                "fnr": "02026100713",
                "fornavn": "LITEN",
                "kjoenn": "M",
                "rolle": "BARN",
              },
            ],
          }
        }
      >
        <Styled(Panel)
          className="neutral"
        >
          <Panel
            border={false}
            className="c7 neutral"
          >
            <div
              className="panel c7 neutral"
            >
              <styled.div>
                <div
                  className="c8"
                >
                  <styled.div>
                    <div
                      className="c9"
                    >
                      <img
                        alt="nav-man-icon"
                        height={50}
                        src="icon-mann.png"
                        width={50}
                      />
                      <styled.div>
                        <div
                          className="c10"
                        />
                      </styled.div>
                      <div
                        data-test-id="panelheader__tittel"
                      >
                        <Undertittel
                          data-test-id="panelheader__tittel__hoved"
                        >
                          <TypografiBase
                            data-test-id="panelheader__tittel__hoved"
                            type="undertittel"
                          >
                            <h2
                              className="typo-undertittel"
                              data-test-id="panelheader__tittel__hoved"
                            >
                              STOR
                               
                              BLYANT
                            </h2>
                          </TypografiBase>
                        </Undertittel>
                        <styled.div>
                          <div
                            className="c11"
                          >
                            <div>
                              ui:form-fnr : 02026100715
                            </div>
                            <div>
                              ui:form-birthdate: 02.02.1961
                            </div>
                          </div>
                        </styled.div>
                      </div>
                    </div>
                  </styled.div>
                  <Styled(Knapp)
                    onClick={[Function]}
                  >
                    <Knapp
                      className="c12"
                      onClick={[Function]}
                    >
                      <KnappBase
                        autoDisableVedSpinner={false}
                        className="c12"
                        disabled={false}
                        htmlType="submit"
                        inaktivKlasseVedDisabled={false}
                        kompakt={false}
                        mini={false}
                        onClick={[Function]}
                        spinner={false}
                        type="standard"
                      >
                        <button
                          className="knapp c12 knapp--standard"
                          disabled={false}
                          onClick={[Function]}
                          type="submit"
                        >
                          <_default
                            color="#0067C5"
                            height="20"
                            width="20"
                          >
                            <svg
                              color="#0067C5"
                              focusable="false"
                              height="20"
                              viewBox="0 0 24 24"
                              width="20"
                            >
                              <path
                                d="M3.516 3.5h16v20h-16zm4-3h8v3h-8zm-6.5 3h22M7.516 7v12m4-12v12m4-12v12"
                                fill="none"
                                stroke="#0067C5"
                                strokeLinecap="round"
                                strokeLinejoin="round"
                                strokeMiterlimit="10"
                              />
                            </svg>
                          </_default>
                          <styled.div>
                            <div
                              className="c10"
                            />
                          </styled.div>
                          <styled.div>
                            <div
                              className="c13"
                            >
                              ui:form-remove
                            </div>
                          </styled.div>
                        </button>
                      </KnappBase>
                    </Knapp>
                  </Styled(Knapp)>
                </div>
              </styled.div>
            </div>
          </Panel>
        </Styled(Panel)>
      </PersonCard>
    </div>
  </styled.div>
</PersonSearch>
`;
